### PR TITLE
Update for m1

### PR DESCRIPTION
**Pull request by Ch0ronomato**
## Description
My work machine is an M2 mac with the apple silicon. Not sure if this is a machine y'all are planning on supporting, but these were the changes I needed to make to environment.yaml to get off the ground.

## Related Issue

- [ ] Closes #
- [ ] Related to #

## Checklist

- [X] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [ ] Included tests that prove the fix is effective or that the new feature works
- [ ] Added necessary documentation (docstrings and/or example notebooks)
- [ ] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Type of change

- [ ] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [X] Maintenance
- [ ] Other (please specify):

### CODE REVIEW

**Comment by maresb** [comment]
Ah, hmm, indeed this is pretty awkward. The proposed changes will break things for those not on Apple Silicon.

Ideally we should be doing something like depend on `libblas` and allow conda's pinning mechanism to install the right thing. I've messed around with it before, but never managed to get good results. I should probably poke at it some more.

**Comment by Ch0ronomato** [comment]
I couldn't get anything either to be totally honest. It doesn't seem like it's a lost battle, but it gets me deving on an m1 for now - though I fully intend for it to be a bit of wasted pr. I did see there was some conda like `# [not osx]` type hints that might be relevant; though adding a `gen_environment.py` that does this resolution for us might be an okay fix

**Review by twiecki — CHANGES_REQUESTED** [review]
This would bake in accelerate instead of MKL, which we probably also don't want.

We figured out how to do this in the conda-forge package. https://github.com/conda-forge/pytensor-suite-feedstock/blob/main/recipe/meta.yaml Not sure it works here? 

Alternatively we could add a different environment.yml for ARM64.

**Comment by Ch0ronomato** [comment]
@twiecki yea, an alternative env file makes sense as well

**Comment by maresb** [comment]
Multiple env files are indeed a straightforward and simple workaround.

Those `# [python_expression]` things are called [preprocessing selectors](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#preprocess-selectors) and unfortunately only apply to the `meta.yaml` Conda recipe specification, not the `environment.yaml` spec.

**Comment by Ch0ronomato** [comment]
Sounds great, I'll go ahead and edit this to be `environment_arm.yml`

**Comment by maresb** [comment]
I'd recommend calling it `environment-osx-arm64.yaml`. (In addition to the `osx-arm64` platform there's also `linux-aarch64` where arm64 and aarch64 are synonymous, and `linux-aarch64` can be used for Docker containers on Apple Silicon.)

File: environment-osx-arm64.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# To use:
+#
+#   $ conda env create -f environment.yml  # `mamba` works too for this command
+#   $ conda activate pytensor-dev
+#
+name: pytensor-dev
+channels:
+  - conda-forge
+dependencies:
+  - python=3.10
```
Comment (maresb) [inline environment-osx-arm64.yml:10]: Why not
```suggestion
  - python=>3.10
```
?

**Review by maresb — COMMENTED** [review]
This approach looks good for now. I think pixi would be able to handle this much more elegantly via features and multienvironments:

https://pixi.sh/latest/features/multi_environment/

But transitioning to pixi is a bit more involved, so let's get this merged for now.

**Review by twiecki — APPROVED** [review]

**Comment by Ch0ronomato** [comment]
Can we get this merged?

**Review by maresb — APPROVED** [review]
Thanks @Ch0ronomato! LGTM

**Pull request merged**